### PR TITLE
rule for createContainer-insteadof-allowZoom

### LINKED ISSUE #95: Use VictoryZoomContainer  createContainer instead of allowZoom prop
https://github.com/patternfly/patternfly-react/pull/4278

### Changes that will be needed:
**Add import:**
import { VictoryZoomContainer } from 'victory-zoom-container';

**Component Changes:**
From:
`<Chart allowZoom>`
  `<ChartGroup allowZoom offset={11} horizontal>`

To:
`<Chart containerComponent={<VictoryZoomContainer />}>`
 `<ChartGroup offset={11} horizontal>`

### PR DESCRIPTION
**Pull request by ddonahue007**
Closes #95 

### createContainer-insteadof-allowZoom [(#4278)](https://github.com/patternfly/patternfly-react/pull/4278)
- **Chart**: Remove `allowZoom` property.
- **Chart**: Add `containerComponent={<VictoryZoomContainer />}` property.
- **ChartGroup**: Remove `allowZoom` property
- Add Import: `import { VictoryZoomContainer } from 'victory-zoom-container';`
#### Examples
In:
```jsx
<Chart allowZoom>
    <ChartAxis />
        <ChartGroup allowZoom offset={11} horizontal>
            "some-chart"
        <ChartGroup/>
<Chart />
```
Out:
```jsx
import { VictoryZoomContainer } from 'victory-zoom-container';

<Chart containerComponent={<VictoryZoomContainer />}>
    <ChartAxis />
        <ChartGroup offset={11} horizontal>
            "some-chart"
        <ChartGroup/>
<Chart />
```

### CODE REVIEW

File: packages/eslint-plugin-pf-codemods/index.js [hunk 1]
```diff
@@ -42,6 +42,8 @@ const rules = {
   "refactor-breakpointmods": require('./lib/rules/refactor-breakpointmods'),
   "datatoolbar-rename-toolbar": require('./lib/rules/datatoolbar-rename-toolbar'),
   "chartVoronoiContainer-remove-allowTooltip": require('./lib/rules/chartVoronoiContainer-remove-allowTooltip'),
+  "createContainer-insteadof-allowZoom": require('./lib/rules/createContainer-insteadof-allowZoom'),
+  "createContainer-insteadof-allowZoom": require('./lib/rules/createContainer-insteadof-allowZoom'),
```
Comment (redallen) [inline packages/eslint-plugin-pf-codemods/index.js:46]: Only need this once. Can you also rename the rule `chart-remove-allowZoom`?

File: packages/eslint-plugin-pf-codemods/lib/rules/createContainer-insteadof-allowZoom.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        ImportDeclaration(node) {
+          const packageImports = getPackageImports(context, ensurePackage);
+          const packageImportNames = packageImports.map(imp => imp.imported.name);
+          const myImports = node.specifiers.map(imp => imp.imported.name);
+          const missingImports = compMap
+              .filter(imp => !packageImportNames.includes(imp)) // not added by consumer
+              .filter(imp => !myImports.includes(imp)) // not added by this rule
+              .join(', ');
+
+          if (packageImports.length == 0 ) {
```
Comment (redallen) [inline packages/eslint-plugin-pf-codemods/lib/rules/createContainer-insteadof-allowZoom.js:39]: Why not keep the code DRY and this `if (packageImports.length == 0 )` logic to the helper?

File: packages/pf-codemods/README.md [hunk 3]
```diff
@@ -841,3 +841,30 @@ Out:
 ```jsx
 <WizardNavItem content="sm" />
 ```
+
+### createContainer-insteadof-allowZoom [(#4278)](https://github.com/patternfly/patternfly-react/pull/4278)
```
Comment (redallen) [inline packages/pf-codemods/README.md:845]: Remember to rename the rule to `chart-remove-allowZoom` here as well

File: packages/eslint-plugin-pf-codemods/index.js (hunk 1 above)
Comment (ddonahue007) [reply packages/eslint-plugin-pf-codemods/index.js]: DOne

File: packages/eslint-plugin-pf-codemods/lib/rules/createContainer-insteadof-allowZoom.js (hunk 2 above)
Comment (ddonahue007) [reply packages/eslint-plugin-pf-codemods/lib/rules/createContainer-insteadof-allowZoom.js:39]: I could do this I was unsure if we wanted to populate the helper with it, thats all. I will move it over now and push it up.

Comment (redallen) [reply packages/eslint-plugin-pf-codemods/lib/rules/createContainer-insteadof-allowZoom.js:39]: Thank you!

Comment (ddonahue007) [reply packages/eslint-plugin-pf-codemods/lib/rules/createContainer-insteadof-allowZoom.js:39]: Actually the main reason I did not do this is that the first check in the helper function is never going to allow the code to trigger.

  if (node.source.value !== package) {
    return;
  }

In this case the import is not tied to a node as it is being used inside an attribute.

**Pull request merged**